raiden/network/rpc/client.py

```diff
@@ -1013,53 +1156,61 @@ def deposit(self, amount):
                 current_balance,
             ))
 
-        transaction_hash = estimate_and_transact(self, self.proxy.deposit, amount)
-
-        self.client.poll(transaction_hash.decode('hex'), timeout=self.poll_timeout)
-        receipt_or_none = check_transaction_threw(self.client, transaction_hash)
-        if receipt_or_none:
-            raise TransactionThrew('Deposit', receipt_or_none)
-
         log.info('deposit called', contract=pex(self.address), amount=amount)
 
-    def opened(self):
-        return self.proxy.opened.call()
+        try:
+            transaction_hash = estimate_and_transact(self, self.proxy.deposit, amount)
+        except:
+            log.critical('deposit failed', contract=pex(self.address))
+            self._check_exists()
+            raise
 
-    def closed(self):
-        return self.proxy.closed.call()
+        try:
+            self.client.poll(
+                transaction_hash.decode('hex'),
+                timeout=self.poll_timeout,
+            )
+        except (InvalidTransaction, JSONRPCPollTimeoutException):
+            log.critical('deposit failed', contract=pex(self.address))
```

Comment:
Why not separate those two exceptions and log a more informative error for each one? Invalid transaction and timeout respectively?